app/src/main/java/com/example/peterchu/watplanner/Views/CourseDetailActivity.java

```diff
@@ -0,0 +1,117 @@
+package com.example.peterchu.watplanner.Views;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.design.widget.CollapsingToolbarLayout;
+import android.support.design.widget.FloatingActionButton;
+import android.support.design.widget.Snackbar;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.View;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.app.ActionBar;
+import android.view.MenuItem;
+
+import com.example.peterchu.watplanner.Constants;
+import com.example.peterchu.watplanner.Database.DatabaseHandler;
+import com.example.peterchu.watplanner.Models.Course.Course;
+import com.example.peterchu.watplanner.Models.Course.CourseDetails;
+import com.example.peterchu.watplanner.Models.Course.CourseDetailsResponse;
+import com.example.peterchu.watplanner.Models.Schedule.CourseSchedule;
+import com.example.peterchu.watplanner.Networking.ApiClient;
+import com.example.peterchu.watplanner.Networking.ApiInterface;
+import com.example.peterchu.watplanner.R;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * An activity representing a single Course detail screen.
+ */
+public class CourseDetailActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_course_detail);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.detail_toolbar);
+        setSupportActionBar(toolbar);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Snackbar.make(view, "Replace with your own detail action", Snackbar.LENGTH_LONG)
+                        .setAction("Action", null).show();
+            }
+        });
+
+        if (savedInstanceState == null) {
+            final DatabaseHandler dbHandler = new DatabaseHandler(this);
+            Course course = dbHandler.getCourse(getIntent().getIntExtra(
+                    CourseDetailFragment.ARG_COURSE_ID, -1));
+            CollapsingToolbarLayout appBarLayout = (CollapsingToolbarLayout) this.findViewById(
+                    R.id.toolbar_layout);
+            if (appBarLayout != null) {
+                appBarLayout.setTitle(course.getName());
+            }
+//            final CourseSchedule courseSchedule = dbHandler.getCourseSchedule(course.getSubject(),
```

Comment:
course schedule db is not imported yet. @timothytong is working on it. once that's done, we can uncomment this